src/neural_network/StraightforwardNeuralNetwork.cpp

```diff
@@ -84,13 +84,13 @@ void StraightforwardNeuralNetwork::train(Data& data, Wait wait, int batchSize, i
     if (!this->isIdle)
         throw std::runtime_error("Neural network is already training");
     this->stopTrainingAsync();
-    wait.startClock();
     this->trainSync(data, wait, batchSize, evaluationFrequency);
 }
 
 void StraightforwardNeuralNetwork::trainSync(Data& data, Wait wait, const int batchSize, const int evaluationFrequency)
```

Comment:
Question: Why do you have two train ?

Reply:
I have a train synchrone and a train asynchrone which runs on a separate thread.